packages/core/src/lib/handler/sms.handler.ts

```diff
@@ -1,11 +1,19 @@
 import { compileTemplate } from '../content/content.engine';
 import { ISmsProvider } from '../provider/provider.interface';
-import { IMessage, ITriggerPayload } from '../template/template.interface';
+import {
+  ChannelTypeEnum,
+  IMessage,
+  ITriggerPayload,
+} from '../template/template.interface';
 
 export class SmsHandler {
   constructor(private message: IMessage, private provider: ISmsProvider) {}
 
   async send(data: ITriggerPayload) {
+    data.$attachments = data.$attachments?.filter((item) =>
+      item.channels?.includes(ChannelTypeEnum.SMS)
```

Comment:
if no channels passed we should behave as if all channels we're selected  here (same on email)

Reply:
So logic should be something like this:
```
item.channels === undefined ||
        item.channels.length === 0 ||
        item.channels?.includes(ChannelTypeEnum.EMAIL)
```
Did I understand you correctly?

Reply:
`item.channels?.length ? item.channels?.includes(ChannelTypeEnum.EMAIL) : true`

Maybe something like this?

Reply:
And we can add a unit test for this it would be amazing 🚀